pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2103,10 +2103,22 @@ func (s *AWSCloud) UpdateTCPLoadBalancer(name, region string, hosts []string) er
 	return nil
 }
 
-// Returns the instance with the specified ID
-func (a *AWSCloud) getInstanceById(instanceID string) (*ec2.Instance, error) {
+// Returns the instance with the specified ID - preserving this incase someone wants to use it
+func (a *AWSCloud) getInstanceByID(instanceID string) (*ec2.Instance, error) {
+	instances, err := a.getInstancesByIDs([]*string{&instanceID})
+	if err != nil {
+		return nil, err
+	}
+	if len(instances) > 1 {
+		return nil, fmt.Errorf("multiple instances found for instance: %s", instanceID)
+	}
+
+	return instances[instanceID], nil
+}
+
+func (a *AWSCloud) getInstancesByIDs(instanceIDs []*string) (map[string]*ec2.Instance, error) {
 	request := &ec2.DescribeInstancesInput{
-		InstanceIds: []*string{&instanceID},
+		InstanceIds: instanceIDs,
 	}
 
 	instances, err := a.ec2.DescribeInstances(request)
```

Comment:
I think you mentioned this somewhere, but did you find documentation on limits on the number of instances we can pass here?  I don't think it will be a problem in practice, but I'd like to be sure!